packages/utils/src/fetchData/index.ts

```diff
@@ -133,11 +137,26 @@ const handleHttpError = (response: Response) => {
 const handleGraphqlError = (error: any) => {
   const errorMessage = error?.message || 'GraphQL error';
   const errorCode = error?.extensions?.code || 400; // Handle extensions (specific to GraphQL)
+  // Log the error for debugging purposes
+  // eslint-disable-next-line no-console
+  console.error('GraphQL Error:', JSON.stringify(error, null, 2));
   // Handle known GraphQL error messages
-  if (errorCode === 'FORBIDDEN') {
+  if (errorCode === 'BAD_USER_INPUT') {
+    throw new ErrorHandler('Bad User Input: The provided input is invalid.', {
+      statusCode: 400,
+    });
+  } else if (errorCode === 'UNAUTHENTICATED') {
+    throw new ErrorHandler('Unauthenticated: Please log in to access this resource.', {
+      statusCode: 401,
+    });
+  } else if (errorCode === 'FORBIDDEN') {
     throw new ErrorHandler('Forbidden access: You do not have the required permissions.', {
       statusCode: 403,
     });
+  } else if (errorCode === 'INTERNAL_SERVER_ERROR') {
```

Comment:
Ik vraag me af, waarom zijn hier strings nodig zoals `'INTERNAL_SERVER_ERROR'`? Kunnen we het niet gewoon bij de integer statuscodes houden? Die zijn in HTTP gestandaardiseerd. We kunnen ook "best of both worlds" hebben met een enum/const van statuscodes, zoals `const INTERNAL_SERVER_ERROR = 500`

Reply:
De reden dat ik een string heb gebruikt, is dat deze errors specifiek GraphQL-errors zijn die dit type fout teruggeven.